### PR TITLE
Continue implementation of Cap Cons: Backward compatibility

### PR DESCRIPTION
**Pull request by turbolent**
Work towards #2091

## Description

- Consider ID capability when resolving path capability final target path; support borrow/check of path capability pointing to ID capability
- Add a temporary `migrateLink` function which replaces a link with a Cap Con

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/stdlib/account.go [hunk 1]
```diff
@@ -3155,6 +3194,146 @@ func newAuthAccountCapabilitiesUnpublishFunction(
...
+					address,
+					borrowType,
+				)
+
+			default:
+				panic(errors.NewUnreachableError())
+			}
+
+			// Publish: overwrite link value with capability
+
```
Comment (turbolent) [inline runtime/stdlib/account.go:3306]: @turbolent Explain why we have to publish also to private path

**Review by dsainati1 — APPROVED** [review]

File: runtime/sema/authaccount.cdc [hunk 2]
```diff
@@ -338,6 +338,14 @@ pub struct AuthAccount {
         /// Returns the capability if one was published at the path.
         /// Returns nil if no capability was published at the path.
         pub fun unpublish(_ path: PublicPath): Capability?
+
+        /// **DEPRECATED**: This function only exists temporarily to aid in the migration of links.
+        /// This function will not be part of the final Capability Controller API.
+        ///
+        /// Migrates the link at the given path to a capability controller.
+        ///
+        /// Does not migrate intermediate links of the chain.
```
Comment (SupunS) [inline runtime/sema/authaccount.cdc:347]: Maybe add a brief description on what the return value is.

**Review by SupunS — APPROVED** [review]

File: runtime/stdlib/account.go (hunk 1 above)
Comment (turbolent) [reply runtime/stdlib/account.go:3310]: Done in 2f9855661ea887ae4d85666a99cf2521483fe87a

File: runtime/sema/authaccount.cdc (hunk 2 above)
Comment (turbolent) [reply runtime/sema/authaccount.cdc:347]: Added in fc6436f0a1f83bbffc7e44502f5102aa8f95ceb1

**Pull request merged**